File: numbers/IsPalindrome.java
// Solution for is_palindrome (integer) in java
package numbers;

public class IsPalindrome {
    public static void main(String[] args) {
        int x = 121;
        System.out.println("testing");
        System.out.println(isPalindrome(x)); // true
        System.out.println(isPalindrome(-x)); // false
        System.out.println(isPalindrome(10)); // false
    }

    public static boolean isPalindrome(int x) {
        if (x < 0 || (x % 10 == 0 && x != 0)) {
            return false;
        }

        int rev = 0;
        int temp = x;
        while (temp != 0) {
            int digit = temp % 10;
            rev = rev * 10 + digit;
            temp /= 10;
        }

        return x == rev;
    }
}

File: numbers/is_palindrome.go
// solution for an integer to be palindromic in Go
package main

import "fmt"

//take an int param and return a bool
func is_palindrome(x int) bool{
 	if x < 0 || (x % 10 == 0 && x != 0) {
		return false
	}
 	rev := 0
 	temp := x
 	for temp > 0 {
		digit := temp % 10
		rev = rev * 10 + digit
		temp /= 10
 	}
	return rev == x
}

func main  () {
	x := 121
	fmt.Println(is_palindrome(x))//true
	fmt.Println(is_palindrome(-x)) //false
	fmt.Println(is_palindrome(10)) //false
}